CONTRIBUTING.md

```diff
@@ -0,0 +1,471 @@
+4. **A code editor** - Visual Studio, VS Code, or Rider recommended
+
+### Verify Your Executable
+
+Ensure your DNCDPRG.EXE has the correct SHA256 hash:
+```
+5f30aeb84d67cf2e053a83c09c2890f010f2e25ee877ebec58ea15c5b30cfff9
+```
+
+You can verify with PowerShell:
+```powershell
+Get-FileHash DNCDPRG.EXE -Algorithm SHA256
+```
+
+Or on Linux/Mac:
+```bash
+sha256sum DNCDPRG.EXE
+```
+
+### Fork and Clone
+
+1. Fork the repository on GitHub
+2. Clone your fork locally:
+```bash
+git clone https://github.com/YOUR-USERNAME/Cryogenic.git
+cd Cryogenic
+```
+
+3. Add the upstream repository:
+```bash
+git remote add upstream https://github.com/OpenRakis/Cryogenic.git
+```
+
+4. Build the project:
+```bash
+cd src
+dotnet build
+```
+
+5. Test that it runs:
+```bash
+dotnet run --Exe /path/to/DNCDPRG.EXE --UseCodeOverride true -p 4096
+```
+
+## Development Environment
+
+### Recommended Tools
+
+- **IDE**: Visual Studio 2022, JetBrains Rider, or VS Code with C# extensions
+- **Debugger**: Built-in IDE debugger for C# code
+- **Ghidra**: For static analysis of assembly code (optional but recommended)
+- **Hex Editor**: For examining data files (optional)
+
+### Project Structure
+
+```
+Cryogenic/
+├── src/
+│   └── Cryogenic/
+│       ├── Program.cs                 # Entry point
+│       ├── DuneCdOverrideSupplier.cs  # Override registration
+│       ├── DriverLoadToolbox.cs       # Driver remapping utilities
+│       ├── Overrides/                 # C# implementations
+│       │   ├── Overrides.cs          # Main override class (partial)
+│       │   ├── VgaDriverCode.cs      # VGA driver overrides
+│       │   ├── MenuCode.cs           # Menu system overrides
+│       │   ├── DialoguesCode.cs      # Dialogue system overrides
+│       │   └── ...
+│       ├── Generated/                 # Auto-generated code (don't edit)
+│       └── Globals/                   # Game state accessors
+├── doc/                              # Documentation and screenshots
+├── dump/                             # Memory dumps and analysis configs
+└── README.md
+```
+
+## Project Architecture
+
+### How Cryogenic Works
+
+Cryogenic uses **Spice86** to run the original DNCDPRG.EXE while selectively replacing assembly routines with C# implementations. This hybrid approach allows:
+
+1. **Incremental rewriting** - Replace one function at a time
+2. **Immediate testing** - Run the game after each change
+3. **Behavioral verification** - Compare C# output with original assembly
+
+### Key Concepts
+
+#### Segments
+
+Memory segments are the foundation of address translation:
+
+- `cs1 = 0x1000` - Main game code
+- `cs2, cs3, cs4` - Mapped driver segments
+- `cs5 = 0xF000` - BIOS/IRQ handlers
+
+**Important**: Never change segment values without auditing ALL `DefineFunction` and `DoOnTopOfInstruction` calls.
+
+#### Override Registration
+
+Overrides are registered in `Overrides.DefineOverrides()` using:
+
+- `DefineFunction(segment, offset, method)` - Replace CALL targets
```

Comment:
Dune often doesn't use calls. Sometimes it edits the stack to modify the return address.

'Replaces identified functions'